mmgen/models/architectures/dalle_mini/modules.py

```diff
@@ -0,0 +1,235 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import torch
+import torch.nn as nn
+from mmcv.cnn.bricks import Linear, build_activation_layer, build_norm_layer
+from mmgen.registry import MODULES
+
+
+@MODULES.register_module()
+class GLU(nn.Module):
+    """GLU variants used to improve Transformer.
+
+    Args:
+        in_out_channels (int): The channel number of the input
+                                and the output feature map.
+        mid_channels (int): The channel number of the middle layer feature map.
+    """
+
+    def __init__(self, in_out_channels, mid_channels):
+        super().__init__()
+        self.norm1 = build_norm_layer(dict(type='LN'), in_out_channels)[1]
+        self.norm2 = build_norm_layer(dict(type='LN'), mid_channels)[1]
+        self.fc1 = Linear(in_out_channels, mid_channels, bias=False)
+        self.fc2 = Linear(in_out_channels, mid_channels, bias=False)
+        self.fc3 = Linear(mid_channels, in_out_channels, bias=False)
+        self.gelu = build_activation_layer(dict(type='GELU'))
+
+    def forward(self, z):
+        """Forward function.
+
+        Args:
+            z (torch.FloatTensor): Input feature map.
+
+        Returns:
+            z (torch.FloatTensor): Output feature map.
+        """
+        z = self.norm1(z)
+        w = self.fc1(z)
+        w = self.gelu(w)
+        v = self.fc2(z)
+        z = self.norm2(w * v)
+        z = self.fc3(z)
+        return z
+
+
+@MODULES.register_module()
+class AttentionBase(nn.Module):
```

Comment:
May this [module](https://github.com/open-mmlab/mmgeneration/blob/master/mmgen/models/architectures/ddpm/modules.py#L91) meets our needs?